packages/router/router.ts

```diff
@@ -3912,15 +3972,88 @@ async function loadLazyRouteModule(
     ...mapRouteProperties(routeToUpdate),
     lazy: undefined,
   });
+
+  return route;
 }
 
-async function callLoaderOrAction(
+function createCallLoaderOrAction(dataStrategy: DataStrategyFunction) {
+  return async (
+    type: "loader" | "action",
+    request: Request,
+    match: AgnosticDataRouteMatch,
+    matches: AgnosticDataRouteMatch[],
+    manifest: RouteManifest,
+    mapRouteProperties: MapRoutePropertiesFunction,
+    basename: string,
+    v7_relativeSplatPath: boolean,
+    opts: {
+      isStaticRequest?: boolean;
+      isRouteRequest?: boolean;
+      requestContext?: unknown;
+    } = {}
+  ): Promise<DataResult> => {
+    let [result] = await dataStrategy({
+      matches: [
+        finesseToAgnosticDataStrategyMatch(match, mapRouteProperties, manifest),
+      ],
+      request,
+      type,
+      defaultStrategy(match) {
+        return callLoaderOrActionImplementation(
+          type,
+          request,
+          match,
+          matches,
+          basename,
+          v7_relativeSplatPath,
+          opts
+        );
+      },
+    });
+
+    return result;
+  };
+}
+
+function finesseToAgnosticDataStrategyMatch(
+  match: AgnosticDataRouteMatch,
+  mapRouteProperties: MapRoutePropertiesFunction,
+  manifest: RouteManifest
+): AgnosticDataStrategyMatch {
+  let loadRoutePromise: Promise<AgnosticDataRouteObject> | undefined;
+
+  if (match.route.lazy) {
+    try {
+      loadRoutePromise = loadLazyRouteModule(
+        match.route,
+        mapRouteProperties,
+        manifest
+      );
+    } catch (error) {
+      loadRoutePromise = Promise.reject(error);
+    }
+  }
+
+  if (!loadRoutePromise) {
+    loadRoutePromise = Promise.resolve(match.route);
+  }
+
+  loadRoutePromise.catch(() => {});
```

Comment:
Could this simplify down to:

```suggestion
  if (match.route.lazy) {
    loadRoutePromise = loadLazyRouteModule(
      match.route,
      mapRouteProperties,
      manifest
    );
  } else {
    loadRoutePromise = Promise.resolve(match.route);
  }
```

- Let `loadLazyRouteModule` rejection propagate out on it's own
- Unsure what we needed the `.catch` no-op for?

Reply:
The catch exists so if a strategy doesn't await all the match promises the error doesn't go uncaught and bring down processes.